### ISSUE TITLE
Failing tests in newer versions of Node

### ISSUE THREAD

**Issue by vegardbb** [jsdom/jsdom#3106]
### Basic info:

- **Node.js version: 14 and up (to my knowledge)**
- **jsdom version: v16 (master branch)**

### Minimal reproduction case

`git checkout` to master and `yarn test-wpt` run the web platform tests.

### Description

The following six tests fail both locally and in one of the Travis builds (the one which runs Node v15, [the four other succeeded](https://github.com/jsdom/jsdom/runs/1482105649))

```
  5063 passing (23m)
  606 pending
  7 failing
  1) web-platform-tests
       cors
         preflight-failure.htm:
     Error: done() called multiple times
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:93:5)
  2) web-platform-tests
       cors
         preflight-failure.htm:
     Uncaught Error: Parse Error: Expected HTTP/
      at Socket.socketOnData (node:_http_client:530:22)
      at Socket.emit (node:events:376:20)
      at addChunk (node:internal/streams/readable:304:12)
      at readableAddChunk (node:internal/streams/readable:279:9)
      at Socket.Readable.push (node:internal/streams/readable:218:10)
      at TCP.onStreamRead (node:internal/stream_base_commons:192:23)
  3) web-platform-tests
       cors
         preflight-failure.htm:
     Error: done() called multiple times
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:93:5)
  4) web-platform-tests
       dom/events
         [expected fail] scrolling/input-text-scroll-event-when-using-arrow-keys.html:
     Uncaught UnhandledPromiseRejection: This error originated either by throwing inside of an async function
without a catch block, or by rejecting a promise which was not handled with .catch().
The promise rejected with the reason "TypeError: element.scrollIntoView is not a function".
  
  5) web-platform-tests
       dom/events
         [expected fail] scrolling/input-text-scroll-event-when-using-arrow-keys.html:
     Error: done() called multiple times
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:93:5)
  6) web-platform-tests
       dom/events
         [expected fail] scrolling/input-text-scroll-event-when-using-arrow-keys.html:
     Error: done() called multiple times
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:93:5)
```

These tests still fail in my branch [replace-fetch](https://github.com/vegardbb/jsdom/tree/replace-fetch), and they need to be resolved before merging with master. May I get some help with some of these?

**Comment by domenic** [comment]
The `done()` called multiple times is probably an issue with our WPT test harness, and worth investigating to find a general fix, or at least the root cause.

The HTTP parsing issue is _probably_ Node tightening their HTTP parser in a way that makes it impossible for us to pass the test, which is unfortunate. That'd need to be disabled, I guess.

The UnhandledPromiseRejection issue should be skipped instead of expected-fail.

**Comment by vegardbb** [comment]
Regarding HTTP Parsing: I've done a little triangulation on the test _preflight-failure.htm_, and I found that the function argument "304" might be what triggers the exception. The test also goes green if only "304" is passed as argument in the script in the test page. For any other combination of commented out arguments, the test does not consistently succeed, but does not consistently fail either.

@domenic What's the status for the build tests?

**Comment by domenic** [comment]
Very strange; that makes it seem more likely to be a Node.js bug to me (although disabling is still a reasonable path).

I don't know what you mean by status for the build tests?

**Comment by vegardbb** [comment]
have the build test been fully migrated over to GitHub Actions from Travis?

**Comment by domenic** [comment]
No; I don't know of any work on that.

**Comment by vegardbb** [comment]
So how do I go about skipping the expected-fail test under "dom/events"?

**Comment by domenic** [comment]
We've been abusing the `timeout` status (instead of `fail`) for that so far.

**Comment by vegardbb** [comment]
Could you point me to an example of using timeouts to ignore a test?

**Comment by domenic** [comment]
https://github.com/jsdom/jsdom/blob/69f6bf689ba240c29c56f04b6e5d855f86e54a30/test/web-platform-tests/to-run.yaml#L563

**Comment by vegardbb** [comment]
As for the last failing test, I am unable to reproduce it locally on my machine:
```
  7) web-platform-tests
       xhr
         abort-during-headers-received.window.html:
     Failed in "XMLHttpRequest: abort() during HEADERS_RECEIVED": 
assert_array_equals: lengths differ, expected array [1, 2, 4] length 3, got [1, 2, 4, 4] length 4
    at Test.<anonymous> (http://web-platform.test:9000/xhr/abort-during-headers-received.window.js:35:13)
    at Test.<anonymous> (http://web-platform.test:9000/resources/testharness.js:2036:22)
    at Test.step (http://web-platform.test:9000/resources/testharness.js:1977:25)
    at http://web-platform.test:9000/resources/testharness.js:2002:35
    at Timeout.task [as _onTimeout] (/home/travis/build/jsdom/jsdom/lib/jsdom/browser/Window.js:391:19)
    at listOnTimeout (node:internal/timers:556:17)
  Error: Failed in "XMLHttpRequest: abort() during HEADERS_RECEIVED": 
  assert_array_equals: lengths differ, expected array [1, 2, 4] length 3, got [1, 2, 4, 4] length 4
  
      at Test.<anonymous> (http://web-platform.test:9000/xhr/abort-during-headers-received.window.js:35:13)
      at Test.<anonymous> (http://web-platform.test:9000/resources/testharness.js:2036:22)
      at Test.step (http://web-platform.test:9000/resources/testharness.js:1977:25)
      at http://web-platform.test:9000/resources/testharness.js:2002:35
      at Timeout.task [as _onTimeout] (lib/jsdom/browser/Window.js:391:19)
      at listOnTimeout (node:internal/timers:556:17)
      at /home/travis/build/jsdom/jsdom/test/web-platform-tests/run-single-wpt.js:197:22
      at http://web-platform.test:9000/resources/testharness.js:2950:22
      at forEach (http://web-platform.test:9000/resources/testharness.js:3653:26)
      at Tests.notify_complete (http://web-platform.test:9000/resources/testharness.js:2947:9)
      at all_complete (http://web-platform.test:9000/resources/testharness.js:2817:22)
      at Tests.complete (http://web-platform.test:9000/resources/testharness.js:2829:13)
      at Tests.notify_result (http://web-platform.test:9000/resources/testharness.js:2806:22)
      at Tests.result (http://web-platform.test:9000/resources/testharness.js:2793:14)
      at cleanup_done (http://web-platform.test:9000/resources/testharness.js:2321:15)
      at Test.cleanup (http://web-platform.test:9000/resources/testharness.js:2260:13)
      at Test.done (http://web-platform.test:9000/resources/testharness.js:2206:14)
      at Test.step (http://web-platform.test:9000/resources/testharness.js:1988:18)
      at http://web-platform.test:9000/resources/testharness.js:2002:35
      at Timeout.task [as _onTimeout] (lib/jsdom/browser/Window.js:391:19)
      at listOnTimeout (node:internal/timers:556:17)
      at processTimers (node:internal/timers:499:7)

```

**Comment by domenic** [comment]
Fixed by #3116!
